src/formic/utils/lmyengine/matrix_builder.h

```diff
@@ -219,6 +219,27 @@ namespace cqmc {
 
       }
 
+    void resetParamNumber(int new_num)
+     {
+     _num_params = new_num;
+     int NumThreads = omp_get_max_threads();
```

Comment:
The is an abstraction to get the number of threads, see src/Concurrency/Info.hpp but the optimization has no guarantee it will run on that many threads so this just isn't an effective way to allocate concurrenct workspace.  It's possible this will work for awhile but it is at least a memory use bomb.

Reply:
#4209 tracks the issue